src/qtt/algorithms/generic.py

```diff
@@ -581,10 +585,47 @@ def weightedCentroid(im, contours, contourIdx, fig=None):
         plt.plot(xyw[0], xyw[1], '.g', markersize=17)
     return xyw
 
+
+def boxcar_filter(signal, kernel_size):
+    """ Perform boxcar filtering on an array.
+    At the edges, the edge value is replicated beyond the edge as needed by the size of the kernel.
+    This is the 'nearest' mode of scipy.ndimage.convolve. For details, see
+    https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.convolve.html?highlight=mode
+
+
+    Args:
+        signal (sequence): a sequence containing the signal to be filtered.
```

Comment:
```suggestion
        signal (array): An array containing the signal to be filtered.
```
The term sequence already as other meanings. Also I would not call a 2D array a 'sequence'

Reply:
Fine with me.